### PR TITLE
slide 5

### PR DESCRIPTION
**Pull request by DenisYaroshevskiy**
avx2 is not done but it's better than it was.

### CODE REVIEW

File: include/eve/detail/shuffle_v2/simd/x86/shuffle_l4_l5.hpp [hunk 1]
```diff
@@ -48,13 +48,34 @@ shuffle_l4_l5_x86_put_u64x2_in_position(P, fixed<G>, wide<T, N> x)
...
+shuffle_l4_l5_x86_slide_less_than_16(P, fixed<G>, wide<T, N> x)
+{
+  constexpr auto no = kumi::tuple {no_matching_shuffle, eve::index<-1>};
+  // Coudn't figure out how to generalize well
+  // only slide left for now
+  // No masking 0s on avx512
+  if constexpr( current_api < avx2 ) return no;
+  else if constexpr ( constexpr auto slide = idxm::is_slide_left(P::idxs) )
+  {
+    static_assert(G == 1, "verifying assumptions");
```
Comment (jfalcou) [inline include/eve/detail/shuffle_v2/simd/x86/shuffle_l4_l5.hpp:69]: if the static_assert is that empty of infos, why not just use the single parameter static_assert and not put a reason ?

File: test/unit/api/regular/shuffle_v2/idxm.cpp [hunk 2]
```diff
@@ -780,6 +799,32 @@ TTS_CASE("is_slide_right")
...
+  };
+
+  test(std::array {0, 1, 2, 3}, 4, 0); // not slide 2, result is unimportant
+  test(std::array {1, 2, 3, 4}, 4, 1);
+  test(std::array {1, 2, 3, 16}, 16, 1);
+  test(std::array {1, we_, 3, 4}, 4, 1);
+  test(std::array {1, we_, 3, 16}, 16, 1);
+  test(std::array {2, 3, 4, we_}, 4, 2);
+  test(std::array {2, 3, 16, we_}, 16, 2);
+  // {2, 3, we_, we_} - is not an important case because it's not slide2 at all.
```
Comment (jfalcou) [inline test/unit/api/regular/shuffle_v2/idxm.cpp:818]: how so ?

File: include/eve/detail/shuffle_v2/simd/x86/shuffle_l4_l5.hpp (hunk 1 above)
Comment (DenisYaroshevskiy) [reply include/eve/detail/shuffle_v2/simd/x86/shuffle_l4_l5.hpp:69]: Well, if it fails for a user, I want them to see that it's internal failure.

Also, now looking at it, this message is bad for that

File: test/unit/api/regular/shuffle_v2/idxm.cpp (hunk 2 above)
Comment (DenisYaroshevskiy) [reply test/unit/api/regular/shuffle_v2/idxm.cpp:818]: It's a one register shuffle. So it's not slide2

Comment (jfalcou) [reply test/unit/api/regular/shuffle_v2/idxm.cpp:818]: OK

File: include/eve/detail/shuffle_v2/simd/x86/shuffle_l4_l5.hpp (hunk 1 above)
Comment (jfalcou) [reply include/eve/detail/shuffle_v2/simd/x86/shuffle_l4_l5.hpp:69]: I'm OK if we go with a better message. I usually try to include "[EVE]" in the message so we know where it comes from

**Review by jfalcou — APPROVED** [review]

**Pull request merged**